### ISSUE TITLE
questions about output space doesn't match all gene space with State Embedding

### ISSUE THREAD

**Issue by Frankie-zju** [ds4cabs/VirtualCellChallenge2025#7]
Excellent Work!
I am trying to use state-embedding as the input of state-transition module, but here is a problem, the output space of StateTranstion has the same dimensions as data.obsm["X_state"], which is 2058 rather than 18080.

So, I am very curious about how you made the configs. My current config  is **on the end of my comment**. 

Besides, I also found that if I want my results go back to all gene counts space, I need a **gene_decoder**, which is a **FinetuneVCICountsDecoder** Class and requires _model_loc="/home/aadduri/vci_pretrain/vci_1.4.2.ckpt"_  and _config="/large_storage/ctc/userspace/aadduri/vci/checkpoint/large_1e-4_rda_tabular_counts_2048/crossds_config.yaml"_.
Do you know where can I find these files?

Looking forward to your reply!

Config.yaml:

data:
  name: PerturbationDataModule
  kwargs:
    toml_config_path: ../../vci_pretrain/starter.toml
    embed_key: X_state
    output_space: all
    pert_rep: onehot
    basal_rep: sample
    num_workers: 4
    pin_memory: true
    n_basal_samples: 1
    basal_mapping_strategy: random
    should_yield_control_cells: true
    batch_col: batch_var
    pert_col: target_gene
    cell_type_key: cell_type
    control_pert: non-targeting
    map_controls: true
    perturbation_features_file: ../../vci_pretrain/ESM2_pert_features.pt
    store_raw_basal: false
    int_counts: false
    barcode: true
  output_dir: null
  debug: true
model:
  name: state
  checkpoint: null
  device: cuda
  kwargs:
    cell_set_len: 128
    blur: 0.05
    hidden_dim: 672
    loss: energy
    confidence_head: false
    n_encoder_layers: 4
    n_decoder_layers: 4
    predict_residual: true
    softplus: true
    freeze_pert: false
    transformer_decoder: false
    finetune_vci_decoder: false
    residual_decoder: false
    batch_encoder: false
    nb_decoder: false
    mask_attn: false
    use_effect_gating_token: false
    use_basal_projection: false
    distributional_loss: energy
    gene_decoder_bool: false
    init_from: null
    transformer_backbone_key: llama
    transformer_backbone_kwargs:
      max_position_embeddings: 128
      hidden_size: 672
      intermediate_size: 2688
      num_hidden_layers: 4
      num_attention_heads: 8
      num_key_value_heads: 8
      head_dim: 84
      use_cache: false
      attention_dropout: 0.0
      hidden_dropout: 0.0
      layer_norm_eps: 1.0e-06
      pad_token_id: 0
      bos_token_id: 1
      eos_token_id: 2
      tie_word_embeddings: false
      rotary_dim: 0
      use_rotary_embeddings: false
training:
  wandb_track: false
  weight_decay: 0.0005
  batch_size: 16
  lr: 0.0001
  max_steps: 40000
  train_seed: 42
  val_freq: 2000
  ckpt_every_n_steps: 2000
  gradient_clip_val: 10
  loss_fn: mse
  devices: 1
  strategy: auto
wandb:
  entity: your_entity_name
  project: state
  local_wandb_dir: ./wandb_logs
  tags:
  - emb_run
name: emb_run
output_dir: ../../competition
use_wandb: true
overwrite: false
return_adatas: false
pred_adata_path: null
true_adata_path: null

**Comment by Frankie-zju** [comment]
Thanks! I found out that I don't need the FinetuneVCICountsDcoder. In the base.py, there is a default LatentToGeneDecoder Class, which will do the same thing. And I found that, it requires the configs of gene_decoder_bool to be True.
